workspace.go

```diff
@@ -208,3 +223,61 @@ func NewWorkspaceVariableResource(v *Variable) *WorkspaceVariableResource {
 		WorkspaceID: fmt.Sprintf("${tfe_workspace.workspace[%q].id}", v.WorkspaceName),
 	}
 }
+
+type WorkspaceTeamAccessResource struct {
+	TeamID      string                 `json:"team_id"`
+	WorkspaceID string                 `json:"workspace_id"`
+	Access      string                 `json:"access,omitempty"`
+	Permissions *TeamAccessPermissions `json:"permissions,omitempty"`
+}
+
+// NewWorkspaceTeamAccessResource takes a Team object and uses it to fill a new WorkspaceTeamAccessResource
+func NewWorkspaceTeamAccessResource(ta *TeamAccess) *WorkspaceTeamAccessResource {
+	return &WorkspaceTeamAccessResource{
+		TeamID:      fmt.Sprintf("${data.tfe_team.%s.id}", ta.TeamName),
+		WorkspaceID: fmt.Sprintf("${tfe_workspace.workspace[%q].id}", ta.WorkspaceName),
+		Access:      ta.Access,
+		Permissions: ta.Permissions,
+	}
+}
+
+type TeamDataResource struct {
+	Name         string `json:"name"`
+	Organization string `json:"organization"`
+}
+
+// AddTeamAccess adds the passed teams to the calling workspace
+func (ws *WorkspaceConfig) AddTeamAccess(teamAccess []TeamAccess, organization string) {
+	if len(teamAccess) == 0 {
+		return
+	}
+
+	ws.Data["tfe_team"] = map[string]interface{}{}
+	ws.Resources["tfe_team_access"] = map[string]interface{}{}
+
+	for _, ta := range teamAccess {
+
+		_, exists := ws.Data["tfe_team"][ta.TeamName]
```

Comment:
Generally `ok` as the second bool argument, for map accesses, type assertions, etc